tools/osquery-testing/test-tables.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Exit if no input file provided.
+[ -z $1 ] && >&2 echo "Error: Input file must be provided" && exit 1
+
+# Read lines from input file.
+cat "$1" | while read -r query
+do
+    # Ignore comments (lines starting with #) and empty lines in the input file.
+    if [ "${query:0:1}" = "#" ] || [ -z "$query" ]; then
+        continue
+    fi
+
+    # Print the query to run.
+    echo sudo osqueryi --line \""$query limit 3"\"
+    echo
+
+    # Run the query ('2>&1' sends stderr to stdout)
+    sudo osqueryi --line "$query limit 3" 2>&1
+    echo
+    echo "---"
```

Comment:
Can we have it also print out the table name and put it at the top? It'll be easier to scan and search if we do that, rather than needing to look at the query itself.

Reply:
That's not necessarily easy to do -- for example, what about queries that have multiple tables?

Would it help if we printed just the query without the rest of the osquery command and the limit portion?

So that it's more like:

```
select * from users cross join user_ssh_keys

        uid = 501
        gid = 20
 uid_signed = 501
 gid_signed = 20
   username = zwass
description = Zach Wasserman
  directory = /Users/zwass
      shell = /bin/zsh
       uuid = 30BAA97B-AE16-42E8-939C-18A4BB516BD0
  is_hidden = 0
       path = /Users/zwass/.ssh/id_ed25519
  encrypted = 1
   key_type =
```

Or, we could change the input format so that the table can be specified. Something like (in `queries.txt`):

```
user_ssh_keys: select * from users cross join user_ssh_keys using (uid)
```

Either of those strategies avoids the (fairly complex) problem of parsing the queries.

Reply:
Let's use the `user_ssh_keys : query` option! It is easier to understand what we are testing.

Reply:
Is it useful to have the whole command line there as well? Seems possibly useful to me, but I don't want to clutter up the results if it's not.

Reply:
I think we can go with whatever is easier there.